app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -85,6 +86,16 @@ def to_s
       title.present? ? title.join(' | ') : 'No Title'
     end
 
+    def total_file_sets
```

Comment:
i'm worried about this implementation on two fronts:

first, i think it's likely to be intolerably slow for collections of any size. it looks to me like it pulls each member work from Fedora, and then pulls each file set from each of those. given the relatively slow turn around on Fedora requests, this could get pathological really quickly! there are definitely optimizations to be had (we can easily count the file_sets in each work from the ids, without retrieving the file set metadata); but i'm still not sure it's a good solution to try to compute this for each request.

second, i don't think the existing implementation will succeed at counting the file sets. it looks like it will miss any file sets that are members transitively (i.e. because they in works that are members of a subcollection of this collection, or because they are in works that are in works that are in this collection).

i think it would be best to discuss this separately from the analytics features bundled here.

Reply:
Removed for now (part of pinned collections feature).

Reply:
i think this might still need to be extracted?